src/popper/runner_host.py

```diff
@@ -229,6 +228,155 @@ def _find_container(self, cid):
         return None
 
 
+class PodmanRunner(StepRunner):
+
+    """Runs steps in podman on the local machine."""
+
+    def __init__(self, init_podman_client=True, **kw):
+        super(PodmanRunner, self).__init__(**kw)
+
+        self._spawned_containers = set()
+
+        if not init_podman_client:
+            return
+
+        try:
+            _, _, self._p_info = HostRunner._exec_cmd(["podman", "info"], logging=False)
+            self._p_version = HostRunner._exec_cmd(["podman", "version"], logging=False)
+        except Exception as e:
+            log.debug(f"Podman error: {e}")
+            log.fail("Unable to connect to podman, is it installed?")
+
+        log.debug(f"Podman info: {pu.prettystr(self._p_info)}")
+
+    def run(self, step):
+        """Executes the given step in podman."""
+        cid = pu.sanitized_name(step.id, self._config.wid)
+        container = self._find_container(cid)
+
+        if not container and self._config.reuse:
+            log.fail(
+                f"Cannot find an existing container for step '{step.id}' to be reused"
+            )
+
+        if container and not self._config.reuse and not self._config.dry_run:
+            cmd = ["podman", "rm", "-f", container]
+            HostRunner._exec_cmd(cmd, logging=False)
+            container = None
+
+        if not container and not self._config.reuse:
+            container = self._create_container(cid, step)
+
+        log.info(f"[{step.id}] podman start")
+
+        if self._config.dry_run:
+            return 0
+
+        self._spawned_containers.add(container)
+
+        cmd = ["podman", "start", "-a", container]
+        _, e, _ = HostRunner._exec_cmd(cmd)
+
+        return e
+
+    def stop_running_tasks(self):
+        """Stop containers started by Popper."""
+        for c in self._spawned_containers:
+            log.info(f"Stopping container {c}")
+            _, ecode, _ = HostRunner._exec_cmd(["podman", "stop", c], logging=False)
+            if ecode != 0:
+                log.warning(f"Failed to stop the {c} container")
+
+    def _find_container(self, cid):
+        """Checks whether the container exists."""
+        cmd = ["podman", "inspect", "-f", "{{.Id}}", cid]
+        _, ecode, containers = HostRunner._exec_cmd(cmd, logging=False)
+
+        if ecode == 125:
+            return None
+
+        if ecode != 0:
+            log.fail(f"podman inspect fail: {containers}")
+
+        return containers.strip()
+
+    def _create_container(self, cid, step):
+        build, _, img, tag, build_ctx_path = self._get_build_info(step)
+
+        if build:
+            log.info(f"[{step.id}] podman build {img}:{tag} {build_ctx_path}")
+            if not self._config.dry_run:
+                pass
+        elif not self._config.skip_pull and not step.skip_pull:
+            log.info(f"[{step.id}] podman build {img}:{tag}")
```

Comment:
this should be `podman pull`

Reply:
Oh that's true since I'm doing a podman pull below, thanks